### PR TITLE
DR-1340 Follow up - update cleanup policy script to use set-IAM-policy 

### PR DESCRIPTION
**Pull request by snf2ye**
We updated this chunk of code in our GitHub actions, but this one-off script was not updated. This is still used when our dev environments hit the 250 IAM policy limit. 

Associated PRs: 
- https://github.com/DataBiosphere/jade-data-repo/pull/616
- https://github.com/DataBiosphere/jade-data-repo/pull/615

How to run:
```
cd jade-data-repo/tools
./cleanupPolicies [GCLOUD PROJECT NAME]
```

### CODE REVIEW

File: tools/cleanupPolicies.sh [hunk 1]
```diff
@@ -4,13 +4,12 @@
...
-for row in $(echo $members | jq -r '.[] | select(startswith("group:policy-"))'); do
-    echo "removing member: ${row}"
-    gcloud projects remove-iam-policy-binding $projectid --member=$row --role=roles/bigquery.jobUser > /dev/null
-done
+# retrieve all IAM policies for data project
+BINDINGS=$(gcloud projects get-iam-policy ${projectid} --format=json)
+# remove any policies that start with group:policy- or deleted:group:policy-
+# group policies are created as a part of our test run and need to be cleared out
+# to avoid hitting 250 IAM policy limit
+OK_BINDINGS=$(echo ${BINDINGS} | jq 'del(.bindings[] | select(.role=="roles/bigquery.jobUser") | .members[] | select(startswith("group:policy-") or startswith("deleted:group:policy-")))')
```
Comment (ddietterich) [inline tools/cleanupPolicies.sh:31]: That is quite a jq expression! Perhaps a comment to explain what it is doing would be useful for a future maintainer.

**Review by ddietterich — APPROVED** [review]
lgtm

File: tools/cleanupPolicies.sh (hunk 1 above)
Comment (snf2ye) [reply tools/cleanupPolicies.sh:31]: JQ expression is courtesy of @fboulnois !

I added more detailed comments - let me know if these make sense, and @fboulnois - please free to edit.

**Review by nmalfroy — APPROVED** [review]
Thanks for putting in the sample JSON too!  That'll be helpful to future us :)

File: tools/cleanupPolicies.sh [hunk 2]
```diff
@@ -1,16 +1,39 @@
...
+# remove any policies for user role BigQuery.JobUsers that start with group:policy- or deleted:group:policy-
+OK_BINDINGS=$(echo ${BINDINGS} | jq 'del(.bindings[] | select(.role=="roles/bigquery.jobUser") | .members[] | select(startswith("group:policy-") or startswith("deleted:group:policy-")))')
+
+# {OK_BINDINGS} traverses the json output from ${BINDINGS}, selecting members to be deleted from policy
+# [from "bindings" array, select member list for role bigquery.jobUser, select only SAM policy members]
+# After del, this leaves us only with the bindings we want to keep (e.g. group:JadeStewards-dev@dev.test.firecloud.org)
+
+# replace the IAM policy, including only members selected in ${OK_BINDINGS}
+echo ${OK_BINDINGS} | jq '.' > policy.json
+gcloud projects set-iam-policy ${projectid} policy.json
```
Comment (fboulnois) [inline tools/cleanupPolicies.sh]:
```suggestion
gcloud projects set-iam-policy ${projectid} policy.json

```

**Review by fboulnois — APPROVED** [review]
lgtm

**Pull request merged**